### DISCUSSION TITLE
Does chatgpt only work in zed if i pay for the api?

### DISCUSSION THREAD

**Discussion by ahmedelami** [zed-industries/zed#9701] [Feature Requests]
i can't use anything from chatgpt4 from my paid plus account. or use the copilot that I pay for(in terms of chatting with, prompting i guess not auto-suggest)?

**Comment by ahmedsaheed** [comment]
Yes - to use the assistant panel or inline assistant for chatting/prompting, you need to add your **OpenAI API key**.
- You can create an API key at: [platform.openai.com/api-keys](https://platform.openai.com/api-keys)
- Make sure your OpenAl account has credits (*not chatGPT4 plus*)
- Having a subscription for ChatGPT4 plus won't work. 
- Having a subscription for another service like GitHub Copilot won't work.

**Reply by gmdias727** [reply]
Why did they make this? I have a paid subscription to OpenAI's ChatGPT, but because of this, I cannot integrate the same ChatGPT into my Zed editor, right?

**Reply by trangcongthanh** [reply]
Because it's 2 different products of OpenAI.
- Subscription to chat.openai.com is a complete chat product built by OpenAI. Not everyone are developer.
- Paid credits for platform.openai.com is API service allow you use other 3rd party products (Like Zed) to interact with ChatGPT models.

**Reply by gmdias727** [reply]
Thank you for clarifying

**Comment by burdiyan** [comment]
These days OpenAI allows using ChatGPT subscription in custom AI agents. Pretty much every agent supports now: OpenCode, Pi, and many others. Could this be enabled in Zed? Using Codex via "External Agents" is so much worse than having native Zed agent working for you.

**Reply by FlyinPancake** [reply]
So what i found is a bit convoluted, but it still works surprisingly well.

You need to set up a LiteLLM proxy that can auth against chatgpt.com. You can find the details in their docs but it's pretty straight forward:

<https://docs.litellm.ai/docs/providers/chatgpt> <https://docs.litellm.ai/docs/proxy/docker_quick_start> 

Once you are set up, configure your providers with a new openapi compatible provider. point it to localhost:4000 and set your master key as the api key. add any number of models you desire and boom

**Comment by star26bsd** [comment]
I support this feature request. While using the vendors' API certainly remains the most professional way of using frontier models, the possibility to use the Codex model of the ChatGPT Pro/Plus subscription would certainly widen the possible user base of Zed.

Please re-open or point us to a more current discussion, thanks!
Stephan
